hoodie-client/src/main/java/com/uber/hoodie/func/payload/BufferedIteratorPayloadFunction.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2018 Uber Technologies, Inc. (hoodie-dev-group@uber.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *          http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.uber.hoodie.func.payload;
+
+import java.util.function.Function;
+
+/**
+ * Function that helps to take an input payload and produce an output payload
+ *
+ * @param <I> input payload data type
+ * @param <O> output payload data type
+ */
+public interface BufferedIteratorPayloadFunction<I, O>
```

Comment:
rename : BufferedIteratorTransform 
(what it does is to transform I to O)

Reply:
I think we should reconsider merging the Payload and the Transform here.. Current way introduces too many classes/interfaces for what we need to achieve.

Reply:
I do agree it introduces extra classes to achieve something more simple...I like the cleaner separation of payload and transformer though.